tools/shared/src/main/scala/org/scalajs/core/tools/linker/LinkedClass.scala

```diff
@@ -161,10 +161,11 @@ object LinkedClass {
 
       // Normal methods
       case m: MethodDef if m.name.isInstanceOf[Ident] =>
-        if (m.body == EmptyTree)
-          abstractMethods += linkedMethod(m)
-        else
+        if (m.body.isDefined) {
           memberMethods += linkedMethod(m)
+        } else {
+          abstractMethods += linkedMethod(m)
+        }
```

Comment:
I don't recall having applied such a rule. For example, it is very common for us to use a `throw` then part without braces.

Maybe you're confusing it with the rule about if/else in *statement* position, which demand that their branches always be on a dedicated line. In other words, no two-liner if/else in statement position.